### PR TITLE
add support for opening shell on remote Windows host

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Add support for running `ansible-test shell --remote windows/<version>` now that SSH is installed.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ansible-test

##### ANSIBLE VERSION
```
devel
```

### CODE REVIEW

**Comment by jborean93** [comment]
@mattclay I'm sure there's more work that needs to be done but I wanted to get started on it. Let me know on your thoughts so far. I was able to connect to a 2016 host by running `ansible-test shell --remote windows/2016` with these changes.

**Comment by mattclay** [comment]
@jborean93 Nice work! SSH tunneling is also working: `Invoke-WebRequest 'http://localhost:8080/'`

One thing we're missing is full tab completion. It's an interesting special case of the completion entries from `remote.txt` and `windows.txt` with `2008` excluded. I recommend creating a `complete_remote_shell` function that handles that, then add a `remote_completion_callback` argument to `add_environments` and pass the custom handler when adding environments to the shell.

See the existing completion related code here:

https://github.com/ansible/ansible/blob/a7d7df145015298627679c9cab67760ba14e1876/test/runner/test.py#L674

https://github.com/ansible/ansible/blob/a7d7df145015298627679c9cab67760ba14e1876/test/runner/test.py#L386

**Comment by mattclay** [comment]
Also, this brings us one step closer to using the http-test-container with Windows tests, since we have a working solution for opening SSH connections to the remote Windows instances.

We shouldn't address that in this PR, but it's something to keep in mind. When Windows tests are running that need the http test container, we're going to want to open a background SSH tunnel (without a terminal) to facilitate the reverse SSH tunneling back to the host running ansible-test.

**Comment by jborean93** [comment]
@mattclay I've tried to add the tab completion for `shell --remote` and it works locally on mine. I haven't worked with argcomplete before so not sure if you prefer a different way to get it done. Let me know your thoughts on the changes.

File: test/runner/test.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def parse_args():
                       help=argparse.SUPPRESS)
 
     add_changes(test, argparse)
-    add_environments(test)
+    add_environments(test, None)
```
Comment (mattclay) [inline test/runner/test.py]: This change doesn't appear to be needed.

**Review by mattclay — COMMENTED** [review]
One minor change

File: test/runner/test.py (hunk 1 above)
Comment (jborean93) [reply test/runner/test.py]: Thanks, accidentally left it there when testing a few things, have fixed it.

**Review by mattclay — APPROVED** [review]

**Comment by jborean93** [comment]
Thanks @mattclay for your help.

**Pull request merged**